src/gui/menu_list.cpp

```diff
@@ -0,0 +1,24 @@
+#include "gui/menu_item.hpp"
+#include "gui/menu_list.hpp"
+#include "gui/menu_manager.hpp"
+
+ListMenu::ListMenu(std::vector<std::string>* items_, int* selected_) : 
+  items(items_), 
+  selected(selected_)
+{
+  int i = 0;
+  for(auto& item : *items) {
+    add_entry(i, item);
+    i++;
+  }
+  add_hl();
+  add_back("OK");
+}
+
+void
+ListMenu::menu_action(MenuItem& item) {
+  *selected = item.get_id();
+  MenuManager::instance().pop_menu();
```

Comment:
If `selected` is null, the behavior here is undefined (Segmentation fault on most systems). Even if erronous, the behavior should be well-defined to avoid needless crashes.

Reply:
I've added a check for the pointer to make sure it's not null.